engine/src/witness/sol.rs

```diff
@@ -104,6 +106,32 @@ impl VoterApi<SolanaIngressTracking> for SolanaIngressTrackingVoter {
 	}
 }
 
+#[derive(Clone)]
+struct SolanaNonceTrackingVoter {
+	client: SolRetryRpcClient,
+}
+
+#[async_trait::async_trait]
+impl VoterApi<SolanaNonceTracking> for SolanaNonceTrackingVoter {
+	async fn vote(
+		&self,
+		_settings: <SolanaNonceTracking as ElectoralSystem>::ElectoralSettings,
+		properties: <SolanaNonceTracking as ElectoralSystem>::ElectionProperties,
+	) -> Result<<<SolanaNonceTracking as ElectoralSystem>::Vote as VoteStorage>::Vote, anyhow::Error>
+	{
+		let (nonce_account, previous_nonce) = properties;
+		let (response_account, response_nonce) =
+			nonce_witnessing::get_durable_nonces(&self.client, vec![nonce_account])
+				.await?
+				.pop()
+				.expect("If the query succeeds, we expect a nonce for the account we queried for");
+
+		assert_eq!(response_account, nonce_account);
+
+		Ok(response_nonce)
```

Comment:
In previous nonce witnessing functionality we only submitted the nonce witnesses to the SC if the witnessed nonce was different than the nonce on the SC. It looks like this change does not have that. I am not sure how the new witnessing works yet but it looks like it will always submit the nonce witness even if the existing nonce on the SC is the same as witnessed nonce.

Reply:
Yeah that's intentional, the SC will now handle this case